dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -1235,228 +1234,102 @@ abstract class DLCWallet
   }
 
   override def findDLC(dlcId: Sha256Digest): Future[Option[DLCStatus]] = {
-    for {
-      dlcDbOpt <- dlcDAO.read(dlcId)
-      contractDataOpt <- contractDataDAO.read(dlcId)
-      offerDbOpt <- dlcOfferDAO.read(dlcId)
-      acceptDbOpt <- dlcAcceptDAO.read(dlcId)
-      (announcements, announcementData, nonceDbs) <- getDLCAnnouncementDbs(
-        dlcId)
-      closingTxFOpt <- dlcDbOpt.map(dlcDb => getClosingTxOpt(dlcDb)) match {
-        case None                 => Future.successful(None)
-        case Some(closingTxIdOpt) => closingTxIdOpt
-      }
-    } yield (dlcDbOpt, contractDataOpt, offerDbOpt) match {
-      case (Some(dlcDb), Some(contractData), Some(offerDb)) =>
-        val totalCollateral = contractData.totalCollateral
-
-        val localCollateral = if (dlcDb.isInitiator) {
-          offerDb.collateral
-        } else {
-          totalCollateral - offerDb.collateral
-        }
-
-        val contractInfo =
-          getContractInfo(contractData,
-                          announcements,
-                          announcementData,
-                          nonceDbs)
-
-        // Only called when safe
-        lazy val (oracleOutcome, sigs) = {
-          val aggSig = dlcDb.aggregateSignatureOpt.get
-          val outcome =
-            contractInfo.sigPointMap(aggSig.sig.toPrivateKey.publicKey)
-
-          val sigs = nonceDbs.flatMap(_.signatureOpt)
-
-          (outcome, sigs)
+    val dlcDbOptF = dlcDAO.read(dlcId)
```

Comment:
parallelize these to try and improve performance, doesn't make a significant difference due to #3213